pkg/event/target/elasticsearch.go

```diff
@@ -228,32 +273,295 @@ func (target *ElasticsearchTarget) Send(eventKey string) error {
 func (target *ElasticsearchTarget) Close() error {
 	if target.client != nil {
 		// Stops the background processes that the client is running.
-		target.client.Stop()
+		target.client.stop()
 	}
 	return nil
 }
 
+func (target *ElasticsearchTarget) checkAndInitClient(ctx context.Context) error {
+	if target.client != nil {
+		return nil
+	}
+
+	clientV7, err := newClientV7(target.args)
+	if err != nil {
+		return err
+	}
+
+	// Check es version to confirm if it is supported.
+	serverSupportStatus, err := clientV7.getServerSupportStatus(ctx)
+	if err != nil {
+		return err
+	}
+
+	switch serverSupportStatus {
+	case ESSUnknown:
+		return errors.New("unable to determine support status of ES (should not happen)")
+
+	case ESSDeprecated:
+		fmt.Println("DEPRECATION WARNING: Support for this version of Elasticsearch will be dropped in a future release. Please upgrade to a version >= 7.x.")
```

Comment:
We know what version it is. Wouldn't it be clearer to pass through the information (getESVersionSupportStatus => getServerSupportStatus => here) and show it to the user?

```suggestion
		fmt.Printf("DEPRECATION WARNING: Support for Elasticsearch version '%s' will be dropped in a future release. Please upgrade to a version >= 7.x.", version)
```

If you agree, the same comment is also valid for the unsupported version error message.

Reply:
@ebozduman Thanks, I've updated the messages with the version info.